packages/dd-trace/test/exporters/agent/writer.spec.js

```diff
@@ -103,6 +103,27 @@ function describeWriter (protocolVersion) {
       writer.flush(done)
     })
 
+    it('should enqueue flushes', (done) => {
```

Comment:
I'm not sure about this one. As a user it would be very unclear that I'm potentially waiting for also other calls to `flush` for my callback to be called. This is made even worse since the signature of the callback accepts an error object, so you could be receiving an error from a different flush, or the other way around.